cpp/src/arrow/compute/exec/hash_join_node_test.cc

```diff
@@ -1061,17 +1066,17 @@ TEST(HashJoin, Random) {
 
     // Print test case parameters
     // print num_rows, batch_size, join_type, join_cmp
-    std::cout << join_type_name << " " << key_cmp_str << " ";
+    std::cout << "Trial " << test_id << ":\n";
+    std::cout << "    " << join_type_name << " " << key_cmp_str << " ";
```

Comment:
I'm a little surprised we have print statements here.  I think we normally avoid this kind of thing for tests, favoring `ARROW_SCOPED_TRACE` instead but since they were here already :shrug:

Reply:
It isn't the preferred style, but it's not necessary to address this here. I've added
https://issues.apache.org/jira/browse/ARROW-14641